### PR TITLE
docs: automated demo deployment

### PR DESCRIPTION
**Pull request by james-milligan**
## This PR

- adds new section to the getting started guide to automatically deploy and port forward to the demo application

### Related Issues

https://github.com/open-feature/open-feature-operator/issues/106

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -2,6 +2,26 @@
...
+## Quick start
+
+### Deploy the demo app
+
+To get started with the operator you can deploy our e2e example using the [playground app](https://github.com/open-feature/playground)
+To deploy the example, run the following command:
+```sh
+make deploy-demo
+```
+This command will deploy our demo app to the `open-feature-demo` namespace, and once it enters a `Ready` state, will start port-forwarding to the deployed `service/open-feature-demo-service`. Once the log line `Forwarding from 127.0.0.1:30000 -> 30000` has printed, the application will be available at [`127.0.0.1:30000`](127.0.0.1:30000).
```
Comment (skyerus) [inline docs/getting_started.md:14]:
```suggestion
This command deploys the demo app to the `open-feature-demo` namespace, and once it enters a `Ready` state, will start port-forwarding to the deployed `service/open-feature-demo-service`. Once the log line `Forwarding from 127.0.0.1:30000 -> 30000` is printed, the application is available at [`127.0.0.1:30000`](127.0.0.1:30000).
```

File: docs/getting_started.md [hunk 2]
```diff
@@ -2,6 +2,26 @@
...
+
+### Uninstall the demo app
+
+To uninstall the demo app from your cluster, simply run the following command:
```
Comment (skyerus) [inline docs/getting_started.md:18]:
```suggestion
To uninstall the demo app from your cluster, run the following command:
```

File: Makefile [hunk 3]
```diff
@@ -139,6 +139,14 @@ deploy-operator:
 	make deploy
 	kubectl wait --for=condition=Available=True deploy --all -n 'open-feature-operator-system'
 
+deploy-demo:
+	kubectl apply -f config/samples/end-to-end.yaml
+	kubectl wait --for=condition=Available=True deploy --all -n 'open-feature-demo'
+	kubectl port-forward -n open-feature-demo service/open-feature-demo-service 30000:30000
```
Comment (skyerus) [inline Makefile:145]: Is this always strictly necessary? I believe this is already handled if you're using a kind cluster

Comment (james-milligan) [reply Makefile:145]: I believe there is the option to have a default exposed port of `30000` when using the `extraPortMappings` configuration in kind but this is not the case by default.  
This also ensures that the one line deployment will work across all clusters (provided there is nothing already bound to the port)

**Review by skyerus — APPROVED** [review]

File: Makefile [hunk 4]
```diff
@@ -139,6 +139,14 @@ deploy-operator:
 	make deploy
 	kubectl wait --for=condition=Available=True deploy --all -n 'open-feature-operator-system'
 
+deploy-demo:
+	kubectl apply -f config/samples/end-to-end.yaml
```
[the remark this answers is not in the document]
Comment (james-milligan) [reply Makefile]: updated

**Review by beeme1mr — APPROVED** [review]

**Pull request merged**